### PR TITLE
Add benchmarks

### PR DESCRIPTION
**Pull request by andersonjoseph**
This PR adds some benchmarks. Here are the results on my machine
```
Benchmark for simple case
Fastify Plugin x 98,947 ops/sec ±2.54% (79 runs sampled)
Fastify Plugin Typescript x 98,539 ops/sec ±1.91% (85 runs sampled)
Fastest is Fastify Plugin Typescript

Benchmark for fastify version checking
Fastify Plugin x 96,928 ops/sec ±2.53% (78 runs sampled)
Fastify Plugin Typescript x 97,231 ops/sec ±1.88% (82 runs sampled)
Fastest is Fastify Plugin Typescript

Benchmark for fastify dependencies checking
Fastify Plugin x 100,273 ops/sec ±1.89% (80 runs sampled)
Fastify Plugin Typescript x 99,840 ops/sec ±1.68% (83 runs sampled)
Fastest is Fastify Plugin

Benchmark for fastify decorators checking
Fastify Plugin x 98,176 ops/sec ±2.01% (80 runs sampled)
Fastify Plugin Typescript x 99,913 ops/sec ±1.75% (84 runs sampled)
Fastest is Fastify Plugin Typescript
```

### CODE REVIEW

File: tsconfig.eslint.json [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{
+  "compilerOptions": {
+    "declaration": true,
+    "target": "es2022",
```
Comment (Eomm) [inline tsconfig.eslint.json]: Does it support node.js 14?

It must follow the fastify's lts matrix: https://fastify.dev/docs/latest/Reference/LTS/#schedule

**Review by metcoder95 — COMMENTED** [review]
I'd follow @Eomm suggestion to test it out on different node versions as well, being the minimum supported by Fastify.

I have some smaller `nit`s but I believe the current state already provides an enough idea of the impact.

Based on the output, it does not deviates much from what I was expecting, almost no difference:

```
Benchmark for simple case
Fastify Plugin x 98,947 ops/sec ±2.54% (79 runs sampled)
Fastify Plugin Typescript x 98,539 ops/sec ±1.91% (85 runs sampled)

Benchmark for fastify version checking
Fastify Plugin x 96,928 ops/sec ±2.53% (78 runs sampled)
Fastify Plugin Typescript x 97,231 ops/sec ±1.88% (82 runs sampled)

Benchmark for fastify dependencies checking
Fastify Plugin x 100,273 ops/sec ±1.89% (80 runs sampled)
Fastify Plugin Typescript x 99,840 ops/sec ±1.68% (83 runs sampled)

Benchmark for fastify decorators checking
Fastify Plugin x 98,176 ops/sec ±2.01% (80 runs sampled)
Fastify Plugin Typescript x 99,913 ops/sec ±1.75% (84 runs sampled)```

The smaller variants shows no valuable difference, then, as said earlier, I believe that the most important topic will become DX 🙂

File: tsconfig.eslint.json (hunk 1 above)
Comment (andersonjoseph) [reply tsconfig.eslint.json]: thanks for the review! I changed the `tsconfig` to be friendly with node.js 14 and I also included node 14 version to the CI checks

**Comment by andersonjoseph** [comment]
> I'd follow @Eomm suggestion to test it out on different node versions as well, being the minimum supported by Fastify.
> 
> I have some smaller `nit`s but I believe the current state already provides an enough idea of the impact.
> 
> Based on the output, it does not deviates much from what I was expecting, almost no difference:
> 
> ```
> Benchmark for simple case
> Fastify Plugin x 98,947 ops/sec ±2.54% (79 runs sampled)
> Fastify Plugin Typescript x 98,539 ops/sec ±1.91% (85 runs sampled)
> 
> Benchmark for fastify version checking
> Fastify Plugin x 96,928 ops/sec ±2.53% (78 runs sampled)
> Fastify Plugin Typescript x 97,231 ops/sec ±1.88% (82 runs sampled)
> 
> Benchmark for fastify dependencies checking
> Fastify Plugin x 100,273 ops/sec ±1.89% (80 runs sampled)
> Fastify Plugin Typescript x 99,840 ops/sec ±1.68% (83 runs sampled)
> 
> Benchmark for fastify decorators checking
> Fastify Plugin x 98,176 ops/sec ±2.01% (80 runs sampled)
> Fastify Plugin Typescript x 99,913 ops/sec ±1.75% (84 runs sampled)```
> 
> The smaller variants shows no valuable difference, then, as said earlier, I believe that the most important topic will become DX 🙂 
> ```

Thanks for the feedback. I'd really appreciate hearing those smaller nits as this is the first time I'm benchmarking a tool in Node.js 😅 most of the code was heavily inspired from other benchmarks in fastify org

**Review by metcoder95 — COMMENTED** [review]
>Thanks for the feedback. I'd really appreciate hearing those smaller nits as this is the first time I'm benchmarking a tool in Node.js 😅 most of the code was heavily inspired from other benchmarks in fastify org

Sure! Honestly are not that big of a deal, mostly personal I would say; like using JS instead of TS for simplicity and more control over the testing, or do the benchmark in some "bundle" manner simulating that is being installed through a package manager (although this one requires ore work).

On the testing part, the combinations are good (maybe I'll add `decorateRequest` which is also missed, and also several `nested` contexts, maybe 2 or 3 levels deep at max) to cover the research.

More into extension rather than any other thing. Great job 🚀

**Pull request merged**